patterns/1-initial/extensions-for-sustainable-growth.md

```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
    - Create criteria for when an extension can be ported to the primary repository, such as adoption of the extension by internal products and demand for the feature.
    - Porting process of the extension to the primary repository will follow more stringent code review guidelines as set by library maintainers.
 
+![Project extensions](../../assets/img/extensions-for-sustainable-growth/extensions-for-sustainable-growth.png)
+
 Following the principles outlined above ensures that:
 
-- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of boilerplate code.
+- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) code.
 - Extensions are discoverable in a repeatable manner to all users of the primary project; just because code doesn't live in the main repository yet does not mean it is not valuable.
 - The maintainer burden is reduced until an extension has demonstrated that it fills an important gap in the primary project.
 - The core project's common code (e.g. base classes and utility functions) can be a starting point for new development that extends project's domain. This avoids the need to port innovative work after-the-fact, thus reducing the overall burden of developing novel features for the project.
 - Developers are more likely to contribute and stay involved in maintenance and building communities for their codebase, which is also good for the health of the overall project ecosystem.
 
 ## Resulting Context
 
-- The project is able to scale with the addition of new features, without adding a maintenance overhead on the primary project repository.
+- The project is able to scale with the addition of new features, without adding maintenance overhead on the primary project repository.
```

Comment:
Is there really zero maintenance work created when using the extensions mechanism?
Or is it rather that the maintenance that is required for this to work seems more fruitful and scales better?

Reply:
since extensions lie outside project, they would not add a maintenance overhead to primary project repository . They will be independent repositories with the new features